cdap-security/src/main/java/io/cdap/cdap/security/tools/AccessTokenGeneratorService.java

```diff
@@ -117,36 +123,35 @@ Codec<AccessToken> getTokenCodec() {
 
   @Override
   protected void startUp() throws Exception {
+    handler.tokenManager.startUp();
     httpService.start();
   }
 
   @Override
   protected void shutDown() throws Exception {
     httpService.stop();
+    handler.tokenManager.stopAndWait();
+  }
+
+  interface TokenGeneratorHttpHandlerFactory {
```

Comment:
Seems overlying complex to have this factory interface just for the token expiration. Can. the expiration be just in cConf? You can have the `main` method to create the injector and instantiate the service from the injector. That seems would be much simpler?